src/perform_step/exponential_rk_perform_step.jl

```diff
@@ -799,6 +799,167 @@ function perform_step!(integrator, cache::EPIRK4s3BCache, repeat_step=false)
+  B[:, 5] = (499125 / (27136 * dt^3)) * R2 - (2187 / (106 * dt^3)) * R3
+  u = uprev + phiv_timestep(dt, A, B; kwargs...)
+
+  # Update integrator state
+  integrator.fsallast = f(u, p, t + dt)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+  integrator.u = u
+end
+
+function perform_step!(integrator, cache::EPIRK5s3Cache, repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack tmp,k,rtmp,rtmp2,A,B,KsCache = cache
+  f.jac(A, uprev, p, t)
+  alg = typeof(integrator.alg) <: CompositeAlgorithm ? integrator.alg.algs[integrator.cache.current] : integrator.alg
+  f0 = integrator.fsalfirst # f(u0) is fsaled
+  kwargs = [(:tol, integrator.opts.reltol), (:iop, alg.iop), (:norm, integrator.opts.internalnorm), 
+    (:adaptive, true), (:caches, KsCache)]
+
+  # Compute U2 horizontally
+  fill!(@view(B[:, 2]), zero(eltype(B)))
+  B[:, 3] .= (55 / (8 * dt)) .* f0
+  B[:, 4] .= (-3025 / (192 * dt^2)) .* f0
+  phiv_timestep!(k, 48dt/55, A, @view(B[:, 1:4]); kwargs...)
+  ## Compute R2
+  @. tmp = uprev + k # tmp is now U2
+  f(rtmp, tmp, p, t + 48dt/55); A_mul_B!(rtmp2, A, k)
+  @. rtmp = rtmp - f0 - rtmp2 # rtmp is now R2
+
+  # Compute U3 horizontally
+  B[:, 2] .= (53/5) .* f0
+  B[:, 3] .= (-648 / (5 * dt)) .* f0
+  B[:, 4] .= (2916 / (5 * dt^2)) .* f0 + (32065 / (1152 * dt^2)) .* rtmp
+  phiv_timestep!(k, 4dt/9, A, @view(B[:, 1:4]); kwargs...)
+  ## Update B matrix using R2
+  B[:, 2] .= f0
+  fill!(@view(B[:, 3]), zero(eltype(B)))
+  B[:, 4] .= (-166375 / (61056 * dt^2)) .* rtmp
+  B[:, 5] .= (499125 / (27136 * dt^3)) .* rtmp
+  ## Compute R3 and update B
+  @. tmp = uprev + k # tmp is now U3
+  f(rtmp, tmp, p, t + 4dt/9); A_mul_B!(rtmp2, A, k)
+  @. rtmp = rtmp - f0 - rtmp2 # rtmp is now R3
+  B[:, 4] .+= (2187 / (106 * dt^2)) .* rtmp
+  B[:, 5] .-= (2187 / (106 * dt^3)) .* rtmp
+  
+  # Update u
+  phiv_timestep!(k, dt, A, B; kwargs...)
+  @. u = uprev + k
+
+  # Update integrator state
+  f(integrator.fsallast, u, p, t + dt)
+  # integrator.k is automatically set due to aliasing
+end
+
+function perform_step!(integrator, cache::EXPRB53s3ConstantCache, repeat_step=false)
+  @unpack t,dt,uprev,f,p = integrator
+  A = f.jac(uprev, p, t)
+  alg = typeof(integrator.alg) <: CompositeAlgorithm ? integrator.alg.algs[integrator.cache.current] : integrator.alg
+  f0 = integrator.fsalfirst # f(uprev) is fsaled
+  kwargs = [(:tol, integrator.opts.reltol), (:iop, alg.iop), (:norm, integrator.opts.internalnorm), (:adaptive, true)]
+
+  # Compute the first group for U2 and U3
+  B = [zeros(f0) f0]
+  K = phiv_timestep([dt/2, 9dt/10], A, B; kwargs...)
+  U2 = uprev + K[:, 1]
+  R2 = f(U2, p, t + dt/2)  - f0 - A*K[:, 1] # remainder of U2
+  U3 = uprev + K[:, 2] # partially
+
+  # Compute the second group for U3
+  B = [zeros(R2) zeros(R2) zeros(R2) R2]
+  K = phiv_timestep([dt/2, 9dt/10], A, B; kwargs...)
+  U3 .+= 216/(25*dt^2) .* K[:, 1] + 8/dt^2 .* K[:, 2]
```

Comment:
For my own future reference, 216 = 8*27, the 8 = 1/2^3 is the factor from the Krylov subspace algorithm because this calculates phi_i / t^i

Reply:
Note that this line isn't fusing because of the scalar operations BTW. Just `@.` it. But optimizations can come later.